pkg/applicationserver/io/packages/loragls/v1/package.go

```diff
@@ -0,0 +1,255 @@
+	"github.com/golang/protobuf/proto"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io"
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io/packages"
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io/packages/loragls/v1/api"
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io/packages/loragls/v1/api/objects"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/events"
+	"go.thethings.network/lorawan-stack/v3/pkg/jsonpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	urlutil "go.thethings.network/lorawan-stack/v3/pkg/util/url"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// PackageName defines the package name.
+const PackageName = "lora-cloud-geolocation-v1"
+
+// GeolocationPackage is the LoRa Cloud Geolocation application package.
+//
+// TODO: Once https://github.com/TheThingsNetwork/lorawan-stack/issues/2533 is merged,
+// remove the custom client usage.
+type GeolocationPackage struct {
+	server   io.Server
+	registry packages.Registry
+	client   *http.Client
+}
+
+// RegisterServices implements packages.ApplicationPackageHandler.
+func (p *GeolocationPackage) RegisterServices(s *grpc.Server) {}
+
+// RegisterHandlers implements packages.ApplicationPackageHandler.
+func (p *GeolocationPackage) RegisterHandlers(s *runtime.ServeMux, conn *grpc.ClientConn) {}
+
+var (
+	errLocationQueryFailed = errors.DefineInternal("location_query_failed", "location query failed")
+	errNoResult            = errors.DefineNotFound("no_result", "no location query result")
+	errNoAssociation       = errors.DefineInternal("no_association", "no association available")
+)
+
+// HandleUp implements packages.ApplicationPackageHandler.
+func (p *GeolocationPackage) HandleUp(ctx context.Context, def *ttnpb.ApplicationPackageDefaultAssociation, assoc *ttnpb.ApplicationPackageAssociation, up *ttnpb.ApplicationUp) (err error) {
+	ctx = log.NewContextWithField(ctx, "namespace", "applicationserver/io/packages/loragls/v1")
+	ctx = events.ContextWithCorrelationID(ctx, append(up.CorrelationIDs, fmt.Sprintf("as:packages:loracloudglsv1:%s", events.NewCorrelationID()))...)
+
+	if def == nil && assoc == nil {
+		return errNoAssociation.New()
+	}
+
+	defer func() {
+		if err != nil {
+			registerPackageFail(ctx, up.EndDeviceIdentifiers, err)
+		}
+	}()
+
+	data, err := p.mergePackageData(def, assoc)
+	if err != nil {
+		return err
+	}
+
+	switch m := up.Up.(type) {
+	case *ttnpb.ApplicationUp_UplinkMessage:
+		return p.sendQuery(ctx, up.EndDeviceIdentifiers, m.UplinkMessage, data)
+	default:
+		return nil
+	}
+}
+
+// Package implements packages.ApplicationPackageHandler.
+func (p *GeolocationPackage) Package() *ttnpb.ApplicationPackage {
+	return &ttnpb.ApplicationPackage{
+		Name:         PackageName,
+		DefaultFPort: 197,
```

Comment:
Declare const

Reply:
The port itself is not used anywhere else - what improved would the const FPort bring ?